operators/utilities.py

```diff
@@ -38,6 +38,43 @@ def select_all(context: Context):
 def deselect_all(context: Context):
     global_data.selected.clear()
 
+def select_invert(context: Context):
+    sketch = context.scene.sketcher.active_sketch
+    if sketch:
+        generator = sketch.sketch_entities(context)
+    else:
+        generator = entities_3d(context)
+    
+    for e in generator:
+        e.selected = not e.selected
+
+def select_extend(context: Context):
+    sketch = context.scene.sketcher.active_sketch
+    if sketch:
+        generator = sketch.sketch_entities(context)
+    else:
+        generator = entities_3d(context)
+    
+    to_select = []
+    for e in generator:
+        if e.is_line() or e.is_curve():
```

Comment:
You can probably use entity.is_segment() alone here

Reply:
done